client/normal_array_of_nillables.go

```diff
@@ -140,3 +172,41 @@ func normalizeNillableCharsArr[R string | []byte, T string | []byte](val []immut
 	}
 	return arr
 }
+
+func areNormalArraysOfNillablesEqual[T comparable](
+	val []immutable.Option[T],
+	f func() ([]immutable.Option[T], bool),
+) bool {
+	if otherVal, ok := f(); ok {
+		return areArraysOfNillablesEqual(val, otherVal)
+	}
+	return false
+}
+
+func areArraysOfNillablesEqual[T comparable](a, b []immutable.Option[T]) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i, v := range a {
+		if v != b[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func areArrayOfNillableBytesEqual(a, b []immutable.Option[[]byte]) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i, v := range a {
+		if v.HasValue() && b[i].HasValue() {
+			if !bytes.Equal(v.Value(), b[i].Value()) {
+				return false
+			}
+		} else if v.HasValue() || b[i].HasValue() {
+			return false
```

Comment:
suggestion: Might be worthwhile to hit this line in test.

Reply:
done